### PR TITLE
timing_load_creds requires POSIX1.2001 due to rusage

### LINKED ISSUE #19838: 3.1-alpha1 - another NonStop compile issue in test/timing_load_creds.c
Another compile problem showed up this morning in test/timing_load_creds.c:

```
c99: running /usr/cmplr/ccomfe -Z8 -ZBL --c99 -ZO2 -Zg3 -ZS/tmp/t1426063457AHFEDG.T -ZR8 -m -Zf -ZT -D_TANDEM_SOURCE -ZNW203 -ZNW220 -ZNW272 -ZNW734 -ZNW770 -ZNW1506 -Zz -Iinclude -I/usr/include -DOPENSSL_BUILDING_OPENSSL -DOPENSSL_VPROC= -D_XOPEN_SOURCE -D_XOPEN_SOURCE_EXTENDED=1 -D_TANDEM_SOURCE -DB_ENDIAN -DNDEBUG test/timing_load_creds.c 
      struct rusage start, end, elapsed;
                    ^
"/home/ituglib/randall/jenkins/.jenkins/workspace/OpenSSL-3.1_Monitor/test/timing_load_creds.c", line 88: error(163):
          incomplete type is not allowed

      struct rusage start, end, elapsed;
                           ^
"/home/ituglib/randall/jenkins/.jenkins/workspace/OpenSSL-3.1_Monitor/test/timing_load_creds.c", line 88: error(163):
          incomplete type is not allowed

      struct rusage start, end, elapsed;
                                ^
"/home/ituglib/randall/jenkins/.jenkins/workspace/OpenSSL-3.1_Monitor/test/timing_load_creds.c", line 88: error(163):
          incomplete type is not allowed

      if (getrusage(RUSAGE_SELF, &start) < 0) {
          ^
"/home/ituglib/randall/jenkins/.jenkins/workspace/OpenSSL-3.1_Monitor/test/timing_load_creds.c", line 161: error(114):
          identifier "getrusage" is undefined

      if (getrusage(RUSAGE_SELF, &start) < 0) {
                    ^
"/home/ituglib/randall/jenkins/.jenkins/workspace/OpenSSL-3.1_Monitor/test/timing_load_creds.c", line 161: error(114):
          identifier "RUSAGE_SELF" is undefined

      if (getrusage(RUSAGE_SELF, &end) < 0) {
          ^
"/home/ituglib/randall/jenkins/.jenkins/workspace/OpenSSL-3.1_Monitor/test/timing_load_creds.c", line 175: error(114):
          identifier "getrusage" is undefined

      if (getrusage(RUSAGE_SELF, &end) < 0) {
                    ^
"/home/ituglib/randall/jenkins/.jenkins/workspace/OpenSSL-3.1_Monitor/test/timing_load_creds.c", line 175: error(114):
          identifier "RUSAGE_SELF" is undefined

      timersub(&end.ru_utime, &start.ru_stime, &elapsed.ru_stime);
      ^
"/home/ituglib/randall/jenkins/.jenkins/workspace/OpenSSL-3.1_Monitor/test/timing_load_creds.c", line 184: error(114):
          identifier "timersub" is undefined
```

### PR DESCRIPTION
**Pull request by t8m**
Fixes #19838

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by rsbeckerca** [comment]
@t8m Trying. That change is based off of 3.2, which has not compiled on NonStop because of a few year old issues. This new compile error crops up when including this change:

```
      if (aligned64(out) && aligned64(in)) {
          ^
"/home/ituglib/randall/openssl-3.0/providers/implementations/ciphers/cipher_aes_gcm_siv_polyval.c", line 49: error(114):
          identifier "uintptr_t" is undefined

      if (aligned64(out) && aligned64(in)) {
          ^
"/home/ituglib/randall/openssl-3.0/providers/implementations/ciphers/cipher_aes_gcm_siv_polyval.c", line 49: error(112):
          expected a ")"

      if (aligned64(out) && aligned64(in)) {
                            ^
"/home/ituglib/randall/openssl-3.0/providers/implementations/ciphers/cipher_aes_gcm_siv_polyval.c", line 49: error(112):
          expected a ")"
```

**Comment by t8m** [comment]
> @t8m Trying.

Can you please cherry-pick the commit on top of the openssl-3.1 checkout and test that instead?

Not saying we shouldn't eventually fix also the build on the master branch but that will require probably much larger effort.

**Comment by rsbeckerca** [comment]
> > @t8m Trying.
> 
> Can you please cherry-pick the commit on top of the openssl-3.1 checkout and test that instead?
> 
> Not saying we shouldn't eventually fix also the build on the master branch but that will require probably much larger effort.

The cherry-pick handled the compile errors. Thanks. I'm running the test suite with hope.

**Comment by rsbeckerca** [comment]
Here are the results:
* The compile is fine.
* 30-test_evp_extra.t fails. This was reported previously on the 3.1 (now 3.2) thread, so I am worried that something is broken here and a fix was not applied. The details of the failure are:
```
    ok 52 - test_ecx_not_private_key
../../util/wrap.pl ../../test/evp_extra_test -context => 139
```
which appears to be a `SIGSEGV`. Running with V=1 is not informative.

**Comment by rsbeckerca** [comment]
The above failure looks exactly like #17537, which was a prior 3.1 (before the renumbering) error that has not been fixed. This is now in the 3.1-alpha thread and is a problem with DLL unloading because of `atexit()` mis-handling.

**Comment by t8m** [comment]
ping for reviews

**Review by beldmit — APPROVED** [review]
LGTM

File: test/timing_load_creds.c [hunk 1]
```diff
@@ -12,14 +12,15 @@
 
 #include <openssl/e_os2.h>
 
-#ifdef OPENSSL_SYS_UNIX
+#if defined OPENSSL_SYS_UNIX
```
Comment (paulidale) [inline test/timing_load_creds.c:15]: We're normally put () around the `OPENSSL_SYS_UNIX`.
Or, better, don't change this?

Comment (t8m) [reply test/timing_load_creds.c:15]: Yeah

File: test/timing_load_creds.c [hunk 2]
```diff
@@ -76,11 +77,12 @@ static void usage(void)
     fprintf(stderr, "          p for private key\n");
     exit(EXIT_FAILURE);
 }
+# endif
 #endif
 
 int main(int ac, char **av)
 {
-#ifdef OPENSSL_SYS_UNIX
+#if defined(_POSIX_VERSION) && _POSIX_VERSION >= 200112L
```
Comment (levitte) [inline test/timing_load_creds.c:85]: The `#else` part should probably be modified, something like this perhaps?

``` C
#else
    fprintf(stderr, "This tool is not supported on this platform for lack of POSIX1.2001 support\n");
    exit(EXIT_FAILURE);
#endif
```

Comment (levitte) [reply test/timing_load_creds.c:85]: ... you missed a spot :wink:

Remove the Windows and VMS specific bits, they're quite unnecessary, really

Comment (t8m) [reply test/timing_load_creds.c:85]: OK, makes some sense too.

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
@beldmit please reconfirm

**Review by beldmit — APPROVED** [review]
LGTM

**Comment by t8m** [comment]
Merged to master and 3.1 branches. Thank you for the reviews.

**Pull request closed without merging**